### PR TITLE
draft version

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

**Review by mfedenko09 — APPROVED** [review]

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java [hunk 1]
```diff
@@ -16,8 +15,10 @@
     private static final String WRONG_DATA_INSERTION = "Будь ласка, введіть коректні дані";
     private static final String WHITESPACE_SEPARATOR = " ";
     private static final String QUIT_SIGN = "q";
-    private BetDao betDao = new BetDaoImpl();
-    private UserDao<User> userDao = new UserDaoImpl<User>();
+    @Inject
+    private BetDao betDao;
+    @Inject
+    private UserDao<User> userDao;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/controller/ConsoleHandler.java]: mentioned about this in prev PR, make code consistent

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                Object objectdao = new Object();
+                if (field.getType() == BetDao.class) {
+                    objectdao = Factory.getBetDao();
+                } else if (field.getType() == UserDao.class) {
+                    objectdao = Factory.getUserDao();
+                }
+                if (objectdao.getClass().isAnnotationPresent(Dao.class)) {
+                    field.set(instance, objectdao);
+                } else {
+                    throw new AnnotationException("Can't find @Dao");
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/lib/Injector.java]: better to add objectDao type name to message to make more informative 
```suggestion
                    throw new AnnotationException("Class doesn't have annotation @Dao");
```

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/lib/Injector.java (hunk 2 above)
Comment (Sashk52) [reply src/main/java/core/basesyntax/lib/Injector.java]: done

File: src/main/java/core/basesyntax/controller/ConsoleHandler.java (hunk 1 above)
Comment (Sashk52) [reply src/main/java/core/basesyntax/controller/ConsoleHandler.java]: I don't understand what have i done?

Comment (Sashk52) [reply src/main/java/core/basesyntax/controller/ConsoleHandler.java]: done

File: src/main/java/core/basesyntax/lib/Injector.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                Object objectdao = new Object();
+                if (field.getType() == BetDao.class) {
+                    objectdao = Factory.getBetDao();
+                } else if (field.getType() == UserDao.class) {
+                    objectdao = Factory.getUserDao();
+                }
+                if (objectdao.getClass().isAnnotationPresent(Dao.class)) {
+                    field.set(instance, objectdao);
+                } else {
+                    throw new AnnotationException("Class doesn't have annotation @Dao");
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/lib/Injector.java]:
```suggestion
                    throw new AnnotationException("Class + objectDao.getType() +  doesn't have annotation @Dao");
```
better to add some info about class as well

**Review by RomanDubovskyi — APPROVED** [review]

File: src/main/java/core/basesyntax/lib/Injector.java (hunk 3 above)
Comment (Sashk52) [reply src/main/java/core/basesyntax/lib/Injector.java]: Do you mean field.getType()? I think objectDao has not method getType().

**Pull request merged**